### PR TITLE
vision batch deployment cli and sdk notebook

### PR DESCRIPTION
**Pull request by gauravrajguru**
# Description
Please review the files in inference folder
cli/foundation-models/system/inference/image-classification/
sdk/foundation-models/system/inference/image-classification/

Other files are just formatted correctly using black since smoke gate was failing.

# Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+    "workspace_ml_client = MLClient(credential, subscription_id, resource_group, workspace_name)\n",
+    "\n",
+    "# the models, fine tuning pipelines and environments are available in the AzureML system registry, \"azureml-preview\"\n",
+    "registry_ml_client = MLClient(\n",
+    "    credential,\n",
+    "    subscription_id,\n",
+    "    resource_group,\n",
+    "    registry_name=\"azureml-preview\",\n",
+    ")\n",
+    "# genrating a unique timestamp that can be used for names and versions that need to be unique\n",
```
Comment (raduk) [inline sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb:86]: genrating -> typo

File: sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,504 @@
+{
+ "cells": [
```
Comment (raduk) [inline sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb:2]: are there any gates for the notebooks/scripts to run them periodically?

**Review by raduk — APPROVED** [review]

File: cli/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.sh [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+# script inputs
+registry_name="azureml-preview"
+subscription_id="<SUBSCRIPTION_ID>"
+resource_group_name="<RESOURCE_GROUP>"
+workspace_name="<WORKSPACE_NAME>"
+
+# This is the model from system registry that needs to be deployed
+model_name="microsoft-beit-base-patch16-224-pt22k-ft22k"
+# using the latest version of the model - not working yet
```
[the remark this answers is not in the document]
Comment (gauravrajguru) [reply cli/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.sh:15]: Passing model as azureml://registries/$registry_name/models/$model_name/label/latest don't work for model deployment command. However, it works for model show.
I used workaround to fetch version number from label to pass in deployment command. Hence, removing this statement.

File: sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb (hunk 1 above)
Comment (gauravrajguru) [reply sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb:86]: Thanks, corrected.

File: sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb (hunk 2 above)
Comment (gauravrajguru) [reply sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb:2]: No, they are not set up as of now. 
I remember, @HrishikeshGeedMS was setting up workflow but was facing challenge in getting enough resources. Please @rjaincc check.

File: cli/foundation-models/system/inference/image-classification/prepare_data.py [hunk 4]
```diff
@@ -34,43 +44,75 @@ def download_and_unzip(dataset_parent_dir: str, is_multilabel_dataset: int):
...
     with open(request_file_name, "w") as request_file:
-        json.dump(request_json, request_file)
\ No newline at end of file
+        json.dump(request_json, request_file)
+
+
+def prepare_data_for_batch_inference(dataset_dir: str, is_multilabel: int = 0):
+    """Prepare image folder and csv file for batch inference.
+
+    This function will move all images to a single image folder and also create a csv
```
Comment (rjaincc) [inline cli/foundation-models/system/inference/image-classification/prepare_data.py:79]: Parameters missing in docstring - let' please add these.

File: cli/foundation-models/system/inference/image-classification/prepare_data.py [hunk 5]
```diff
@@ -34,43 +44,75 @@ def download_and_unzip(dataset_parent_dir: str, is_multilabel_dataset: int):
...
+    file with images in base64 format.
+    """
+    image_list = []
+
+    csv_file_name = "ml_image_list.csv" if is_multilabel == 1 else "mc_image_list.csv"
```
Comment (rjaincc) [inline cli/foundation-models/system/inference/image-classification/prepare_data.py:84]: nit: better to call it "image_classification_multilabel_lis.csv" and "image_classification_multiclass_list.csv"

File: cli/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (rjaincc) [reply cli/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.sh]: @gauravrajguru - Let's please make the same change in the FT notebooks as well.

File: sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb (hunk 2 above)
Comment (rjaincc) [reply sdk/python/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.ipynb:2]: @gauravrajguru , @raduk  - These workflows are the ones which are being ported to ADO.

File: cli/foundation-models/system/inference/image-classification/prepare_data.py (hunk 4 above)
Comment (gauravrajguru) [reply cli/foundation-models/system/inference/image-classification/prepare_data.py:99]: Added parameters descriptions.

File: cli/foundation-models/system/inference/image-classification/prepare_data.py (hunk 5 above)
Comment (gauravrajguru) [reply cli/foundation-models/system/inference/image-classification/prepare_data.py]: Changed.

File: cli/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (gauravrajguru) [reply cli/foundation-models/system/inference/image-classification/image-classification-batch-endpoint.sh]: I'll handle it in another PR. We will need to do changes in notebooks and CLIs (total 4) as well. Let this PR complete, will help in updating the links.

File: sdk/python/foundation-models/system/finetune/image-classification/multiclass-classification/hftransformers-fridgeobjects-multiclass-classification.ipynb [hunk 6]
```diff
@@ -220,7 +216,9 @@
     "aml_registry_model_name = \"microsoft-beit-base-patch16-224-pt22k-ft22k\"\n",
     "model_version = \"1\"\n",
```
Comment (rjaincc) [inline sdk/python/foundation-models/system/finetune/image-classification/multiclass-classification/hftransformers-fridgeobjects-multiclass-classification.ipynb:217]: @gauravrajguru  - I mean here. We are specifying the version manually, right? I meant, let's bring this in sync with how we have it in other notebook, extract the latest version and use that.

Comment (gauravrajguru) [reply sdk/python/foundation-models/system/finetune/image-classification/multiclass-classification/hftransformers-fridgeobjects-multiclass-classification.ipynb]: Updated it.

**Pull request merged**